### DISCUSSION TITLE
MultiSelect should hover first filtered result and Enter should do the search

### DISCUSSION THREAD

**Discussion by ftzi** [mantinedev/mantine#3092] [Feature requests]
It's not currently possible in [MultiSelect](https://github.com/mantinedev/mantine/blob/master/src/mantine-core/src/MultiSelect/MultiSelect.tsx) to type and press enter to select the first result; it's required to hover a result with the cursor.

1) I suggest a new property such as `hoverOnType` to automatically select the first result, so the user can press Enter to select it. 

2) After the item is selected and no item is hovered, pressing Enter again should not call [`event.preventDefault()`](https://github.com/mantinedev/mantine/blob/master/src/mantine-core/src/MultiSelect/MultiSelect.tsx), so a `<form>`'s `<Button type='submit'>` could catch this and to the Search on Enter press. Or, a new prop like `onSubmit` that would be called instead.

**Comment by rtivital** [comment]
1. I'm fine with `hoverOnType` prop. You are welcome to submit a PR to add it to Select, MultiSelect and Autocomplete components.
2. Currently, I'm not ready to change this logic, as it seems to work correctly – form is submitted when the dropdown is closed.

**Reply by tianhuil** [reply]
@rtivital: thank you for the quick response!  For #2, I'm used to hitting "enter" to submit a form.  It is the only component in this search bar and I don't want the user to have to click a "submit" button.  Is there a way to get this behavior?

**Reply by ftzi** [reply]
Instead of changing the logic, it could have another prop like `submitOnEnter`. It wouldn't change the existing behavior but it would allow this functionality for those who opt-in. As description, it could be something like `"Submits form on Enter if no item is hovered"` (and maybe `and if it's focused`)
